### DISCUSSION TITLE
ot-br: on IPv6-only network with no source of prefix, OnLinkPrefix for LAN side derived from extPanID

### DISCUSSION THREAD

**Discussion by rretanubun** [openthread/openthread#9070] [Q&A] [answered]
Hello,

Trying to make sense of what we are observing...

# Setup
- A TBR directly connected to A PC via ethernet, the PC eth interface is configured to supports only IPv6 and there is no other source that generates an IPv6 prefix.

# Procedure
- Boot TBR and observe that it obtains a `Link-local ip address` (e.g. fe80:....) and an a new `Local on-link prefix` (e.g. `fdab:d017:704b:340d::/64`)
- Use `dataset init new` -> `dataset commit active` to form a new network
- We observe that a new `Local on-link prefix` is now regenerated based on `extPanID`
 the relevant code path (I think) are here:
https://github.com/openthread/openthread/blob/main/src/core/border_router/routing_manager.cpp#LL2444C1-L2456C27
https://github.com/openthread/openthread/blob/main/src/core/border_router/routing_manager.cpp#L2137-L2150

# Question
- If deriving `Local on-link prefix` from `extPanID` expected behavior in this case?
- If so, what is the rationale for coupling these two?

Thank you for everyone's time!

**Comment by abtink** [comment, marked as answer]
> If deriving Local on-link prefix from extPanID expected behavior in this case?

Yes It is. This is specified in Thread Specification in section "9.6.2".

> If so, what is the rationale for coupling these two?

To make sure all BRs connected to same Thread mesh advertise same on-link prefix.

**Reply by rretanubun** [reply]
Thanks for the clarification Abtin!
